frontend/src/components/molecules/AddTransaction.tsx

```diff
@@ -1,15 +1,67 @@
 import moment from 'moment';
-import React from 'react';
+import React, { ReactElement, useReducer } from 'react';
 import styled from 'styled-components';
 import { useAuthState } from '../../store/contexts/auth';
 import { useTransactionDispatch } from '../../store/contexts/transactions';
 import { TransactionActions } from '../../store/reducers/transactions';
 import Collapsable from '../atoms/Collapsable';
+import SnackBarContainer from '../atoms/SnackBarContainer';
 import Form from './Form';
 
+const initialState = { snax: <div />, content: '' };
+
+interface IState {
+  snax: ReactElement;
+  content: string;
+  clicker?: () => void;
+}
+
+interface IAction {
+  type: 'clear' | 'good' | 'bad';
```

Comment:
Argh, already wrote a lot of comments, but updating the file removed them all 🙃 

This doesn't have to be an object.

```tsx
type SnAction = 'clear' | 'good' | 'bad';
```